### PR TITLE
Add release workflow

### PR DESCRIPTION
**Pull request by mkuznyetsov**
Use GH Actions to perform release of che-operator

### CODE REVIEW

**Comment by tolusha** [comment]
The correct version of the operator-sdk is 0.15.2 right now.
We are working on https://github.com/eclipse/che/issues/18349 to fix some problems with that

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+          set -e
+
+          OP_SDK_DIR=/opt/operator-sdk
+          mkdir -p $OP_SDK_DIR
+          wget https://github.com/operator-framework/operator-sdk/releases/download/v0.10.0/operator-sdk-v0.10.0-x86_64-linux-gnu -O $OP_SDK_DIR/operator-sdk 
+          chmod +x $OP_SDK_DIR/operator-sdk
+          export PATH="$PATH:$OP_SDK_DIR"
+
+          # get base32 util from che-release project
```
Comment (nickboldt) [inline .github/workflows/release.yml:45]: do we still need this if we can just install the real base32?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+          fi
+          
+          # default robot account on quay does not have permissions for application repos
+          # so we provide credentials for accounts that do
+          export QUAY_USERNAME_K8S=${{ secrets.QUAY_ECLIPSE_CHE_OPERATOR_KUBERNETES_PASSWORD }}
+          export QUAY_PASSWORD_K8S${{ secrets.QUAY_ECLIPSE_CHE_OPERATOR_KUBERNETES_PASSWORD }}
+          export QUAY_USERNAME_OS=${{ secrets.QUAY_ECLIPSE_CHE_OPERATOR_OPENSHIFT_USERNAME }}
+          export QUAY_USERNAME_OS=${{ secrets.QUAY_ECLIPSE_CHE_OPERATOR_OPENSHIFT_PASSWORD }}
+
+          # git checkout ${CHE_VERSION}
```
Comment (nickboldt) [inline .github/workflows/release.yml:72]: why are these lines commented out?

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+          # ./make-release.sh ${CHE_VERSION} --push-olm-files
+          
+          # perform extra checkouts to ensure branches exist locally
+          git checkout ${BRANCH}
```
Comment (nickboldt) [inline .github/workflows/release.yml:76]: if branch already exists will this fail? should we have a `|| true` on it?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Set up Python 3.6
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+      - name: Install operator courier
+        run: |
+          python -m pip install --upgrade pip
+          # pip install operator-courier
```
Comment (nickboldt) [inline .github/workflows/release.yml:26]: you're not installing operator-courier here. why commented out?

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+      - name: Release operator
+        run: |
+          CHE_VERSION=${{ github.event.inputs.version }}
+          BRANCH=${CHE_VERSION%.*}.x
+
+          #preinstall operator-courier and operator-sdk
+          pip install operator-courier==2.1.7
```
Comment (nickboldt) [inline .github/workflows/release.yml:34]: should all the python related installations be done at the same time in the previous step?

**Comment by tolusha** [comment]
We switched to a new operator-sdk version (0.15.2) in master and might switch to even newer in context of https://github.com/eclipse/che/issues/18349

File: .github/workflows/release.yml (hunk 3 above)
Comment (mkuznyetsov) [reply .github/workflows/release.yml:74]: it shouldn't

File: .github/workflows/release.yml (hunk 2 above)
Comment (mkuznyetsov) [reply .github/workflows/release.yml:70]: fixed

File: .github/workflows/release.yml (hunk 1 above)
Comment (mkuznyetsov) [reply .github/workflows/release.yml:41]: I'll use the same for consitency for now

**Review by tolusha — APPROVED** [review]

**Pull request merged**